py/desiutil/test/test_redirect.py

```diff
@@ -0,0 +1,148 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""Test desiutil.redirect.
+
+Note that desiutil is not MPI-aware, so we only run MPI unit tests if the
+'RUN_MPI_TESTS' environment variable is set.  This way we can run those
+tests on machines where MPI is available, but skip them when running on
+CI environments, etc.
+
+"""
+import sys
+import os
+import re
+import unittest
+import tempfile
+
+import subprocess as sp
+
+
+test_serial_source = """
+import sys
+import os
+
+from desiutil.redirect import stdouterr_redirected
+
+def generate_output(error=False):
+    if error:
+        raise RuntimeError("Error!")
+    else:
+        for i in range(5):
+            print("{}".format(i))
+
+filename = sys.argv[1]
+with_error = (sys.argv[2] == "1")
+
+with stdouterr_redirected(to=filename):
+    if with_error:
+        generate_output(error=True)
+    else:
+        generate_output()
+
+"""
+
+test_mpi_source = """
+from mpi4py import MPI
+import sys
+import os
+
+from desiutil.redirect import stdouterr_redirected
+
+def generate_output(error=False):
+    if error:
+        raise RuntimeError("Error!")
+    else:
+        for i in range(5):
+            print("{}".format(i))
+
+filename = sys.argv[1]
+with_error = (sys.argv[2] == "1")
+
+with stdouterr_redirected(to=filename, comm=MPI.COMM_WORLD):
+    if with_error and MPI.COMM_WORLD.rank == MPI.COMM_WORLD.size - 1:
+        generate_output(error=True)
+    else:
+        generate_output()
+
+"""
+
+
+class TestRedirect(unittest.TestCase):
+    """Test desiutil.redirect"""
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Add `tearDownClass(cls)` to clean up this temporary directory.

Reply:
Done